crates/pipeline_manager/src/db/mod.rs

```diff
@@ -587,249 +540,276 @@ impl Storage for ProjectDB {
-            let project_id = row.get::<_, Option<i64>>(6).map(ProjectId);
-            let mut descr = ConfigDescr {
-                config_id,
-                project_id,
+            let program_id = row.get::<_, Option<Uuid>>(5).map(ProgramId);
+            let created_secs: i64 = row.get(6);
+            let created_naive = NaiveDateTime::from_timestamp_millis(created_secs * 1000)
+                .ok_or_else(|| {
+                    AnyError::msg(format!(
+                        "Invalid timestamp in 'pipeline.created' column: {created_secs}"
+                    ))
+                })?;
+
+            let mut descr = PipelineDescr {
+                pipeline_id,
+                program_id,
                 version: Version(row.get(1)),
                 name: row.get(2),
                 description: row.get(3),
                 config: row.get(4),
-                pipeline: None,
                 attached_connectors: Vec::new(),
+                created: DateTime::<Utc>::from_utc(created_naive, Utc),
+                port: row.get::<_, Option<i16>>(7).unwrap_or(0) as u16,
+                shutdown: row.get(8),
             };
+            descr.attached_connectors = self.json_to_attached_connectors(row.get(9)).await?;
 
-            descr.attached_connectors = self.json_to_attached_connectors(row.get(7)).await?;
-            if let Some(pipeline_id) = pipeline_id {
-                descr.pipeline = Some(self.get_pipeline(pipeline_id).await?);
-            }
+            Ok(descr)
+        } else {
+            Err(DBError::UnknownPipeline(pipeline_id).into())
+        }
+    }
+
+    async fn get_pipeline_by_name(&self, name: String) -> AnyResult<PipelineDescr> {
+        let row = self
+            .pool
+            .get()
+            .await?
+            .query_opt(
+                "SELECT p.id, version, description, p.config, program_id,
+                        created, port, shutdown,
+                COALESCE(json_agg(json_build_object('name', ac.name,
+                                                    'connector_id', connector_id,
+                                                    'config', ac.config,
+                                                    'is_input', is_input))
+                                FILTER (WHERE ac.name IS NOT NULL),
+                        '[]')
+                FROM pipeline p
+                LEFT JOIN attached_connector ac on p.id = ac.pipeline_id
+                WHERE p.name = $1
+                GROUP BY p.id
+                ",
+                &[&name],
+            )
+            .await?;
+
+        if let Some(row) = row {
+            let pipeline_id = PipelineId(row.get(0));
+            let program_id = row.get::<_, Option<Uuid>>(4).map(ProgramId);
+            let created_secs: i64 = row.get(5);
+            let created_naive = NaiveDateTime::from_timestamp_millis(created_secs * 1000)
```

Comment:
Is this just a super defensive check? We create these timestamps right, so why might they be invalid?

Reply:
I took this from `get_pipeline_by_id` which already did it like this to be the same. Will factor it into a common function.